src/components/ProjectSettingsTabs/AliasRow.tsx

```diff
@@ -0,0 +1,159 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import {
+  SegmentedControl,
+  SegmentedControlOption,
+} from "@leafygreen-ui/segmented-control";
+import { Subtitle } from "@leafygreen-ui/typography";
+import { Accordion } from "components/Accordion";
+import { SpruceFormProps } from "components/SpruceForm";
+
+const VariantInput: Record<string, string> = {
+  Regex: "variant",
+  Tags: "variantTags",
+} as const;
+
+const TaskInput: Record<string, string> = {
+  Regex: "task",
+  Tags: "taskTags",
+} as const;
+
+// Extract index of the current field via its ID
+const getIndex = (id: string): number => {
+  if (!id) return null;
+
+  const stringIndex = id.substring(id.lastIndexOf("_") + 1);
+  const index = Number(stringIndex);
+  return Number.isInteger(index) ? index : null;
+};
+
+export const AliasRow: SpruceFormProps["ObjectFieldTemplate"] = ({
+  disabled,
+  formData,
+  idSchema,
+  properties,
+  readonly,
+  uiSchema,
+}) => {
+  const [, alias, variant, variantTags, task, taskTags] = properties;
+  const isDisabled = disabled || readonly;
+  const accordionTitle = uiSchema["ui:accordionTitle"];
+
+  const index = getIndex(idSchema.$id);
+
+  const [variantInput, setVariantInput] = useState(
+    formData?.variant ? VariantInput.Regex : VariantInput.Tags
+  );
+  const [taskInput, setTaskInput] = useState(
+    formData?.variant ? TaskInput.Regex : TaskInput.Tags
+  );
+
+  // Clear form when toggling between segmented control options
+  const clearForm = (selectedSegment: string) => {
+    switch (selectedSegment) {
+      case VariantInput.Regex:
+        variantTags.content.props.onChange([]);
+        break;
+      case VariantInput.Tags:
+        variant.content.props.onChange("");
+        break;
+      case TaskInput.Regex:
+        taskTags.content.props.onChange([]);
+        break;
+      case TaskInput.Tags:
+        task.content.props.onChange("");
+        break;
+      default:
+    }
+  };
+
+  return (
+    <Accordion
+      title={`${accordionTitle} ${index !== null ? index + 1 : ""}`}
+      defaultOpen={!isDisabled}
+      titleTag={AccordionTitle}
+      contents={
+        <div>
+          {alias.content}
+          <TaskRegexContainer>
+            <StyledSegmentedControl
+              label="Variant"
+              id="variant-input-control"
+              value={variantInput}
+              onChange={(value) => {
+                setVariantInput(value);
+                clearForm(value);
+              }}
```

Comment:
how would you feel about not clearing it between toggles? On the plugins page where there's a toggle between custom ticket creation versus default ticket creation, instead of clearing the form, I leave it there in case the user changes their mind and wants to come back to it, and instead make sure to only save the appropriate one when the form is saved.

Reply:
Definitely would prefer not clearing it—I went with this approach because it was significantly easier to implement 😅 I could try looking into it a bit more, or we could create a ticket as an improvement for later. Open to either!

Reply:
ah I definitely hear that! I would definitely suggest a separate ticket as opposed to looking into to it now :)

Reply:
EVG-16214